packages/peregrine/lib/talons/CartPage/PriceAdjustments/ShippingMethods/useShippingForm.js

```diff
@@ -32,9 +32,24 @@ export const useShippingForm = props => {
 
     const [
         setShippingAddress,
-        { called: setShippingAddressCalled, loading: isSetShippingLoading }
+        {
+            called: setShippingAddressCalled,
+            error: setShippingAddressError,
+            loading: isSetShippingLoading
+        }
     ] = useMutation(setShippingAddressMutation);
 
+    let derivedError;
+    if (setShippingAddressError) {
+        if (setShippingAddressError.graphQLErrors) {
+            derivedError = setShippingAddressError.graphQLErrors
+                .map(({ message }) => message)
+                .join(', ');
+        } else {
+            derivedError = setShippingAddressError.message;
+        }
```

Comment:
I smell a utility function!

Reply:
We are probably not going to render Apollo errors going forward, since they can't be trusted to be human readable. I have moved this logic to a new `FormError` component which should reduce the boiler plate though, but you still gotta handroll this logic if you don't like the presentation of that component.